### PR TITLE
=htc simplify decompressor/deflate infrastructure

### PR DESCRIPTION
**Pull request by jrudolph**
The idea is to replace the handlers in `DecompressorParsingLogic` with simple parameters of the `Inflate` step.

### CODE REVIEW

**Comment by jrudolph** [comment]
These changes should be backported to the copy in akka-stream.

**Comment by jrudolph** [comment]
The coding implementation classes were still public (accidentally, I'd say). So, I made them private to allow the binary incompatible changes. Strictly taken this is a violation of our binary compatibility guarantees. I'd say that it is unlikely that other code relies on that code but we cannot know of course. We could also postpone the actual changes to 10.1.0 but maybe that's overkill for these bits of infrastructure.

**Comment by jrudolph** [comment]
PLS BUILD

File: akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala [hunk 1]
```diff
@@ -86,74 +90,68 @@ private[http] object DeflateCompressor {
...
      * https://www.ietf.org/rfc/rfc1951.txt (no-wrap)
      */
-    private def examineAndBuildInflater(bytes: ByteString): Inflater =
-      bytes.headOption.filter(b ⇒ (b & 0x0F) != 8)
-        .fold(new Inflater())(_ ⇒ new Inflater(true))
-
-    override def afterInflate = inflateState
-    override def afterBytesRead(buffer: Array[Byte], offset: Int, length: Int): Unit = {}
+    private def examineAndBuildInflater(bytes: ByteString): Inflater = {
+      val wrapped = (bytes.head & 0x0F) == 0x08
```
Comment (johanandren) [inline akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala]: The old one took into account that the `ByteString` could be empty, but now we are sure it isn't?

Comment (jrudolph) [reply akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala]: Yes, exactly. It's an invariant of `ParseStep.parse` that the byte reader will contain at least a single byte to read.

**Comment by jrudolph** [comment]
Added a section to upcoming release notes about the new incompatibility.

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by tom-walford** [comment]
Thanks for getting this all sorted @jrudolph :)

**Pull request merged**